java/runtime/src/main/java/io/ray/runtime/io/MemoryBuffer.java

```diff
@@ -0,0 +1,958 @@
+package io.ray.runtime.io;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.base.Preconditions;
+import java.nio.BufferOverflowException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.ReadOnlyBufferException;
+
+/**
+ * A class for operations on memory managed by Ray. The buffer may be backed by heap memory (byte
+ * array) or by off-heap memory. If the memory is off-heap memory, it will be the memory allocated
+ * by ray object store.
+ *
```

Comment:
Maybe mention that this class can auto grow on write operations.

Reply:
Yes, should document that.